packages/transformers/html/src/dependencies.js

```diff
@@ -154,6 +146,70 @@ export default function collectDependencies(asset: MutableAsset, ast: AST) {
       return node;
     }
 
+    if (tag === 'script' && attrs.src) {
+      let context = new Set(asset.env.context);
+      context.delete('module');
+      context.delete('script');
+      context.add(attrs.type === 'module' ? 'module' : 'script');
+
+      let loc = node.loc
+        ? {
+            filePath: asset.filePath,
+            start: node.loc.start,
+            end: node.loc.end,
+          }
+        : undefined;
+
+      let outputFormat = 'global';
+      if (attrs.type === 'module' && asset.env.scopeHoist) {
+        outputFormat = 'esmodule';
+      } else {
+        delete attrs.type;
+      }
+
+      // If this is a <script type="module">, and not all of the browser targets support ESM natively,
+      // add a copy of the script tag with a nomodule attribute.
+      let copy: ?PostHTMLNode;
+      if (
+        outputFormat === 'esmodule' &&
+        asset.env.engines.browsers &&
+        !asset.env.supports('esmodules')
+      ) {
+        let attrs: {[string]: string, ...} = {...node.attrs};
+        copy = {...node, attrs};
+        delete attrs.type;
+        attrs.nomodule = '';
+        attrs.src = asset.addURLDependency(attrs.src, {
+          // Keep in the same bundle group as the HTML.
+          isAsync: false,
+          isEntry: false,
+          isIsolated: true,
+          env: {
+            context,
+            outputFormat: 'global',
+            loc,
+          },
+        });
+
+        seen.add(copy);
+      }
+
+      attrs.src = asset.addURLDependency(attrs.src, {
```

Comment:
A thought: both of these dependencies count against the parallel request limit, but only one will actually be downloaded. How should we represent that in the graph? 🤔